v2/tests/log_sink_etw_int/log_sink_etw_int.c

```diff
@@ -0,0 +1,683 @@
+
+            start_parse_events();
+
+            // act
+            log_sink_etw.log_sink_log(event_log_level, NULL, __FILE__, __FUNCTION__, captured_line, "test_event");
+
+            // assert
+            wait_for_event_count(expected_event_count);
+
+            // if we are not expecting an event, wait for some time just to make sure no event is seen
+            if (expected_event_count == 0)
+            {
+                (void)printf("Waiting for %.02f to make sure no events are published ...", (double)TEST_WAITTIME_NO_EVENT / 1000);
+                Sleep(TEST_WAITTIME_NO_EVENT);
+            }
+
+            stop_trace(trace_session_handle);
+            stop_parse_events();
+
+            POOR_MANS_ASSERT(parsed_event_count == expected_event_count);
+
+            if (expected_event_count > 0)
+            {
+                const char* expected_event_name;
+                switch (event_log_level)
+                {
+                default:
+                    expected_event_name = NULL;
+                    break;
+
+                case LOG_LEVEL_CRITICAL:
+                    expected_event_name = "LogCritical";
+                    break;
+                case LOG_LEVEL_ERROR:
+                    expected_event_name = "LogError";
+                    break;
+                case LOG_LEVEL_WARNING:
+                    expected_event_name = "LogWarning";
+                    break;
+                case LOG_LEVEL_INFO:
+                    expected_event_name = "LogInfo";
+                    break;
+                case LOG_LEVEL_VERBOSE:
+                    expected_event_name = "LogVerbose";
+                    break;
+                }
+
+                POOR_MANS_ASSERT(strcmp(parsed_events[0].event_name, expected_event_name) == 0);
+                POOR_MANS_ASSERT(strcmp(parsed_events[0].content, "test_event") == 0);
+                POOR_MANS_ASSERT(strcmp(parsed_events[0].file, __FILE__) == 0);
+                POOR_MANS_ASSERT(strcmp(parsed_events[0].func, __FUNCTION__) == 0);
+                POOR_MANS_ASSERT(parsed_events[0].line == captured_line);
+            }
+        }
+    }
+}
+
+static void log_sink_etw_log_with_context_with_properties(void)
+{
+    // arrange
+    generate_trace_session();
+
+    TRACEHANDLE trace_session_handle = start_trace(TRACE_LEVEL_VERBOSE);
+
+    int captured_line = __LINE__;
+
+    LOG_CONTEXT_HANDLE log_context;
+    LOG_CONTEXT_CREATE(log_context, NULL,
+        LOG_CONTEXT_STRING_PROPERTY(gigi, "duru"),
+        LOG_CONTEXT_PROPERTY(int8_t, prop1, 42),
+        LOG_CONTEXT_PROPERTY(uint8_t, prop2, 43),
+        LOG_CONTEXT_PROPERTY(int16_t, prop3, 44),
+        LOG_CONTEXT_PROPERTY(uint16_t, prop4, 45),
+        LOG_CONTEXT_PROPERTY(int32_t, prop5, 46),
+        LOG_CONTEXT_PROPERTY(uint32_t, prop6, 47),
+        LOG_CONTEXT_PROPERTY(int64_t, prop7, 48),
+        LOG_CONTEXT_PROPERTY(uint64_t, prop8, 49)
```

Comment:
>uint64_t [](http://example.com/codeflow?start=29&length=8)

due to how the values are picked from memory, may I suggest values that fill all the bytes of the type, with different values?

In this case, a 0x12345678ABCDEF01 value would fill all the 8 bytes with different values.

Reply:
For all types, here and everywhere.

Reply:
Done.